### DISCUSSION TITLE
Suggestion: Custom Colour Scheme options

### DISCUSSION THREAD

**Discussion by FranticFairy** [OpenTTD/OpenTTD#12920] [Features]
Currently, the Colour Scheme tab has a few pre-defined categories, automatically hides categories with 0 examples (Like the Maglev/Monorail options if you don't have any), but can be made very big and has a scroll bar. Additionally, it can leave some things to be desired, like if you have a Driving Trailer, it takes one of the engine categories, rather than the passenger coach category.

Perhaps more could be done with this?

I propose allowing GRF Developers to define their own categories, and allow them to define what category something belongs to in terms of liveries. This way, developers can do small things like shift around what categories things go into for recolouring (putting Driving Trailers in the Passenger Coach category, for example), or go wild and make an entirely new set of categories for their GRF.

~~Sub-categories could be a thing similar to how passenger coaches are done currently. One main category (Passenger Coach) that has different categories that trigger based on the **_parent_**'s category.~~ New idea, perhaps instead just allow setting what category something is in through use of switches, just like what you can currently do for graphics. More control and options this way.

Possible uses for this include separate colour selectors for cargo, express and mixed-use locomotives, or separating 3rd rail from Overhead Units... Just. More choice for playing around with colours.

**Comment by Ragin1924** [comment]
If I were to pick a way for this to work, I'd ideally like it to work on a tag system. So if I tag something with Livery_Class: Diesel/Electric/EMU/DMU, it goes into that category, but then maybe have the ability to add modifier categories. E.g. Livery_Class: Diesel, Passenger, Large, or Electric, 3rd rail, long distance, etc. Ideally it'd also have a fall back, so in that first case, if the player sets the livery for a large, passenger diesel, it recieves that colour scheme, but if the player hasn't, it falls back onto the previous tag. So if no large, passenger diesel livery is set, it falls back to the passenger diesel, then diesel, and then default, in whatever order the GRF author puts them in when compiling. Ideally I'd also like for tags to be standardised, so the passenger/diesel/EMU/3rd rail tag works across GRFs, so you don't end up with duplicate company colour options.
